code/game/turfs/open/lava.dm

```diff
@@ -46,6 +48,8 @@
 	if(!smoothing_flags)
 		update_appearance()
 
+	new /obj/effect/abstract/particle_holder (src, particle_type)
```

Comment:
I am concerned about the overhead of this. there is a good amount of lava on the map. Could you profile lava Init before/after this change?

Similar concern, how does this impact client performance near a lava pool? could we batch this somehow to reduce the clientside load?

Reply:
This is me standing in the middle of lava near the necrop gate
![image](https://user-images.githubusercontent.com/7501474/229741119-7f4300d6-8eef-41d0-9bd7-7f3fbb9fd90e.png)

This is me standing in an area without lava
![image](https://user-images.githubusercontent.com/7501474/229741229-8c64bd8a-3682-4808-9789-514c27d147f7.png)

Below is the profiling I did for with and without larticles, n=3
[profile for larticles.zip](https://github.com/tgstation/tgstation/files/11146562/profile.for.larticles.zip)

Reply:
Lava went from around 830 to 2200 from a glance

Reply:
1.2 seconds really kinda sucks. nowhere near enough to block it, but I'd like to reduce it if possible.
Would you mind taking the "base sample" client profile in the same place, but without your changes? I'd like to make sure nothing's fuckey (also init_profiler.json from those rounds would be useful to optimize your changes, if you have it.)

Reply:
Melbert's idea is good but I would rather you do it with a fixed counter rather than `prob(20)` for consistentcy. 1.2 seconds is really not ok

Reply:
Btw why does this need particle_holder in the first place? Lava doesn't move

Reply:
Talked with mothblocks about the init time stuff some.

I think it's reasonable to directly set the particles var in cases like this. We're never gonna want extra behavior here, and if we keep things to one a type, and use holders for extra shit, then we'll do ok. Especially to accomplish effects like this.

I can optimize particle holders, but it's still gonna be like 7k atom news on lavaland (more actually now that rivers work), which is not great no matter what I do

We could also do something like `if((x + ((y % 2) * 3))) % 6 == 0)` to ensure they land predictably, but offset (could use a few rand()s to introduce controlled randomness). This is maybe me flexing a bit, the prob would work fine too

Reply:
![image](https://user-images.githubusercontent.com/58055496/229949357-175b491e-7793-4ed8-96b0-9015ef7efccb.png)
Maybe we could make the holders singletons? or pull from a pool?

Reply:
we'd want more then one, because otherwise the effect would just duplicate I think

Reply:
Good point, I'll make like a pool or something they can pick from

Reply:
would this not have an insane overhead client side as you need to deal with every lava on screen at once, Wouldn't it be better to have the lava particles exist as a giant particle holder on a plane and have lava turfs render target it so you only see the particles on those locations

Reply:
It’s not great, although I’ve already managed to optimize it to an acceptable level. I’m not entirely sure about what you’re describing but I’ll look into it